include/sound/sof/ipc4_ext_manifest.h

```diff
@@ -0,0 +1,122 @@
+/* SPDX-License-Identifier: ((GPL-2.0 WITH Linux-syscall-note) OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2021 Intel Corporation. All rights reserved.
+ */
+
+/*
+ * Extended manifest is a place to store metadata about firmware, known during
+ * compilation time - for example firmware version or used compiler.
+ * Given information are read on host side before firmware startup.
+ * This part of output binary is not signed.
+ */
+
+#ifndef __SOF_IPC4_FIRMWARE_EXT_MANIFEST_H__
+#define __SOF_IPC4_FIRMWARE_EXT_MANIFEST_H__
+
+/* In ASCII  $AE1 */
+#define SOF_IPC4_EXT_MAN_MAGIC_NUMBER    0x31454124
+
+#define SOF_IPC4_MAX_MODULE_NAME_LEN     8
+#define SOF_IPC4_MAX_FW_BINARY_NAME      8
+#define SOF_IPC4_DEFAULT_HASH_SHA256_LEN 32
+#define SOF_IPC4_18_FW_HDR_OFFSET	0x2000
+
+struct sof_ipc4_ext_manifest_hdr {
+	u32 id;
```

Comment:
Can this be named as `magic` like it is with the IPC3 header?

Reply:
We could rename it and it is better to change on the firmware side, too. [struct fw_ext_man_cavs_header](https://github.com/thesofproject/rimage/blob/9a26e4558094f19f3f7becd89eb9e8a9a9dd82b9/src/include/rimage/cavs/cavs_ext_manifest.h#L228)